File: c/2574.0_Left_and_Right_Sum_Differences.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* leftRightDifference(int* nums, int numsSize, int* returnSize) {
    int total = 0;
    for (int i = 0; i < numsSize; i++) {
        total += nums[i];
    }    

    int left_sum = 0;
    for (int i = 0; i < numsSize; i++) {
        int x = nums[i];
        nums[i] = abs(left_sum * 2 + x - total);
        left_sum += x;
    }
    *returnSize = numsSize;
    return nums;
}

File: leetcode-cn/2574.0_Left_and_Right_Sum_Differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        s = sum(nums)
        ans = []
        lefts, rights = 0, s
        for x in nums:
            rights -= x
            ans.append(abs(lefts - rights))
            lefts += x
        return ans


class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        total = sum(nums)
        left_sum = 0
        for i, x in enumerate(nums):
            nums[i] = abs(left_sum * 2 + x - total)
            left_sum += x
        return nums
